### PR TITLE
[FFM-1848]: Functional and performance testing

### PR DESCRIPTION
**Pull request by akiraqb**

### CODE REVIEW

File: src/test/java/io/harness/cf/client/api/TestUtils.java [hunk 1]
```diff
@@ -28,4 +33,40 @@ public static String getJsonResource(String location) throws IOException, URISyn
...
+      list.add(df);
+    }
+    //    System.out.println(list);
+    return list;
+  }
+
+  public FeatureConfig GetFeatureConfigFromFile() throws Exception {
+    try {
+      String relativePath =
+          "./src/test/resources/local-test-cases/basic_bool_string_for_repository.json";
```
Comment (davejohnston) [inline src/test/java/io/harness/cf/client/api/TestUtils.java:57]: Does this file already exist ?

File: src/main/java/io/harness/cf/client/api/StorageRepository.java [hunk 2]
```diff
@@ -98,6 +99,37 @@ public Optional<FeatureConfig[]> getCurrentAndPreviousFeatureConfig(@NonNull Str
...
 
+  public FeatureSnapshot getFeatureSnapshot(@NonNull String identifier) {
+    Gson gson = new Gson();
+    final String flagKey = formatFlagKey(identifier);
+    final String pFlagKey = formatPrevFlagKey(identifier);
+
+    FeatureConfig pFlag = (FeatureConfig) cache.get(pFlagKey);
+    FeatureConfig cFlag = (FeatureConfig) cache.get(flagKey);
+
+    if (cFlag != null) {
```
Comment (davejohnston) [inline src/main/java/io/harness/cf/client/api/StorageRepository.java:110]: should there be a null check for pFlag as well here?  I'm assuming there is an assumption that if we dont have the current then we wont have the previous.

**Review by davejohnston — APPROVED** [review]
This looks good, a minor comment just to check, but if youre happy go for it.

File: src/test/java/io/harness/cf/client/api/TestUtils.java (hunk 1 above)
Comment (akiraqb) [reply src/test/java/io/harness/cf/client/api/TestUtils.java:57]: yes

File: src/main/java/io/harness/cf/client/api/StorageRepository.java (hunk 2 above)
Comment (akiraqb) [reply src/main/java/io/harness/cf/client/api/StorageRepository.java:110]: if the previous flag does not exist we still want to package that to the snapshot even if its null.

**Pull request merged**